scripts/dts/gen_defines.py

```diff
@@ -195,11 +195,19 @@ def write_node_comment(node):
 """
 
     if node.description:
-        # Indent description by two spaces
-        s += "\nDescription:\n" + \
-            "\n".join("  " + line for line in
-                      node.description.splitlines()) + \
-            "\n"
+        # We used to put descriptions in the generated file, but
+        # devicetree bindings now have pages in the HTML
+        # documentation. Let users who are accustomed to digging
+        # around in the generated file where to find the descriptions
+        # now.
+        #
+        # Keeping them here would mean that the descriptions
+        # themselves couldn't contain C multi-line comments, which is
+        # inconvenient when we want to do things like quote snippets
+        # of .dtsi files within the snippets themselves, or otherwise
+        # include the characters "*/" in order.
+        s += ("\n(Descriptions have moved to the Devicetree Bindings Index\n"
```

Comment:
FWIW I had tried to include comments in descriptions and it broke YAML parsing.  Might have been due to things that were within the comments, though, since it appears the ones you added work.

Also this will eliminate any in-header description for bindings that aren't in any generated documentation.

Reply:
> Also this will eliminate any in-header description for bindings that aren't in any generated documentation.

I don't think that's true. If a description from a binding makes it into the header, it's because the relevant node had a matching compatible. Which means that compatible's binding will be in the bindings index.

Unless I'm missing something, that can't happen.

Reply:
I was thinking about out-of-tree bindings: application specific, or module-specific, or other situations where somebody might not have a comprehensive Devicetree Bindings Index (or wouldn't know to generate one).  But I haven't dug into what the tooling provides in that situation.

Reply:
> I was thinking about out-of-tree bindings: application specific, or module-specific, or other situations where somebody might not have a comprehensive Devicetree Bindings Index

Fair enough, but I think at that point they know the binding and this is a worthwhile tradeoff until we get better tooling for inspecting the DT.

Reply:
I don't have a problem with the change; the generated comment still provides the full path to the binding, so people should have enough of a clue.